discussions-store/discussions-store-api/src/main/java/org/xwiki/contrib/discussions/store/DiscussionContextMetadataStoreService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.discussions.store;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.contrib.discussions.domain.DiscussionContext;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Interface dedicated for saving context metadata.
+ * We separate this interface from {@link DiscussionContextStoreService} as the metadata might be kept in separate
+ * objects.
+ *
+ * @version $Id$
+ * @since 2.2
+ */
+@Role
+public interface DiscussionContextMetadataStoreService
+{
+    /**
+     * Read the metadata for the given discussion context.
+     * This method will modify the given discussion context to set the metadata values.
+     *
+     * @param discussionContext the context for which to read the metadata.
+     * @return {@code true} if some metadata has been found and the discussion context has been modified.
+     */
+    boolean readMetadata(DiscussionContext discussionContext);
```

Comment:
`load` might express better that the operation has side-effects on the discussion context

Reply:
Indeed, better naming. I'll change it.